packages/plugin-typescript/src/lib/runner/ts-runner.ts

```diff
@@ -0,0 +1,25 @@
+import {
+  type Diagnostic,
+  createProgram,
+  getPreEmitDiagnostics,
+} from 'typescript';
+import { loadTargetConfig } from './utils.js';
+
+export type DiagnosticsOptions = {
+  tsconfig: string;
+};
+
+export async function getTypeScriptDiagnostics({
+  tsconfig,
+}: DiagnosticsOptions): Promise<readonly Diagnostic[]> {
+  const { fileNames, options } = await loadTargetConfig(tsconfig);
+  try {
+    const program = createProgram(fileNames, options);
+    // @TODO use more fine-grained helpers like getSemanticDiagnostics instead of getPreEmitDiagnostics
+    return getPreEmitDiagnostics(program);
+  } catch (error) {
+    throw new Error(
+      `Can't create TS program in getDiagnostics. \n ${(error as Error).message}`,
```

Comment:
We have a [utility function](https://github.com/code-pushup/cli/blob/main/packages/utils/src/lib/errors.unit.test.ts) that takes care of safely converting unknown errors to a string:

```suggestion
      `Can't create TS program in getDiagnostics.\n${stringifyError(error)}`,
```